transformer_engine/common/include/transformer_engine/multi_tensor.h

```diff
@@ -17,13 +17,58 @@
 extern "C" {
 #endif
 
+/*!  \brief Computes L2 norm for a list of contiguous tensors, providing the option to
+ *          immediately exit the kernel based on the value of the 'noop' tensor.
+ *
+ * \warning   This API is **experimental** and subject to change.
+ * \warning   Argument device_id is deprecated and will be removed in a future release.
+ *
+ *  \param[in]     chunk_size              Number of tensors per kernel launch,
+                                           limited by total size of kernel args.
+ *  \param[in]     noop_flag               Noop tensor.
+ *  \param[in]     tensor_lists            2D array of contiguous input tensors.
+ *  \param[in]     num_tensor_lists        Size (dim0) of tensor_lists.
+ *  \param[in]     num_tensors_per_list    Size (dim1) of tensor_lists.
+ *  \param[in]     output                  Scratch space. Required size grows with number of inputs.
+ *  \param[in]     output_per_tensor       Fixed size auxilliary scratch space.
+ *  \param[out]    ret                     L2 norm of all inputs.
+ *  \param[out]    ret_per_tensor          L2 norm for each tensor.
+ *  \param[in]     per_tensor              Whether to calculate per tensor or cumulative norm.
+ *  \param[in]     max_chunks_per_tensor   Maximum number of chunks in any input tensor.
+ *  \param[in]     device_id               [DEPRECATED] CUDA device ID for this operation.
+ *  \param[in]     stream                  CUDA stream used for this operation.
+ */
 void nvte_multi_tensor_l2norm_cuda(int chunk_size, NVTETensor noop_flag, NVTETensor **tensor_lists,
                                    const size_t num_tensor_lists, const size_t num_tensors_per_list,
                                    NVTETensor output, NVTETensor output_per_tensor, NVTETensor ret,
                                    NVTETensor ret_per_tensor, int per_tensor,
                                    int max_chunks_per_tensor, const int device_id,
                                    cudaStream_t stream);
 
+/*!  \brief Computes L2 norm for a list of contiguous tensors after unscaling.
+ *          Unscaling is only done for computing the L2 norm, tensors are not updated.
+ *          Providing the option to immediately exit the kernel based on the
+ *          value of the 'noop' tensor.
```

Comment:
We should separate the brief description from the details:
```suggestion
/*! \brief Compute L2 norm for a list of tensors after unscaling.
 *
 *  Unscaling is only done for computing the L2 norm. The tensors themselves are not updated.
```